File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> ans = new Stack<>();

        for(char c : s.toCharArray()){

            if( c=='(' || c=='[' || c=='{' ){
                ans.push(c);
            }
            else{
                if(ans.isEmpty()){
                    return false;
                }

                char top = ans.peek();
                if(c == ')' && top != '(' || c == '}' && top != '{' || c == ']' && top != '['){
                    return false;
                }else{
                    ans.pop();
                }
            }
        }

        return ans.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st ;

        int num = s.size();
        for(int i=0 ; i<num ; i++){
            if( s[i] == '(' || s[i] == '[' || s[i] == '{'){
                st.push( s[i] );
            }

            else if(!st.empty() && ((s[i] == ')' && st.top() == '(') ||
                                    (s[i] == ']' && st.top() == '[') || 
                                    (s[i] == '}' && st.top() == '{'))){
                st.pop();
            }
            else{
                return false ;
            }
        }

        return st.empty();
    }
};